modules/apps/commerce/commerce-inventory-service/src/main/java/com/liferay/commerce/inventory/internal/upgrade/v3_0_0/CommerceInventoryWarehouseUpgradeProcess.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.inventory.internal.upgrade.v3_0_0;
+
+import com.liferay.commerce.inventory.internal.upgrade.v3_0_0.util.CommerceInventoryWarehouseTable;
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.upgrade.util.UpgradeProcessUtil;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.LocalizationUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Andrea Sbarra
+ */
+public class CommerceInventoryWarehouseUpgradeProcess extends UpgradeProcess {
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		alter(
+			CommerceInventoryWarehouseTable.class,
+			new AlterColumnType("name", "STRING null"),
+			new AlterColumnType("description", "STRING null"));
+
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				"update CIWarehouse set name = ?,description = ? where " +
+					"CIWarehouseId = ?");
+			Statement s = connection.createStatement();
+			ResultSet resultSet = s.executeQuery(
+				"select CIWarehouseId, companyId, name, description from " +
+					"CIWarehouse")) {
+
+			while (resultSet.next()) {
+				String languageId = UpgradeProcessUtil.getDefaultLanguageId(
+					resultSet.getLong("companyId"));
+
+				preparedStatement.setString(
+					1,
+					LocalizationUtil.getXml(
+						HashMapBuilder.put(
+							languageId, resultSet.getString("name")
+						).build(),
+						languageId, "name"));
+
+				String description = resultSet.getString("description");
+
+				if (!Validator.isBlank(description)) {
```

Comment:
@luca-pellizzon I'm not sure how to handle this case, let me know what do you think

Reply:
I don't understand the issue, if the description is empty there's nothing to handle here. Once somebody will put a description it will be already the localized one.

Reply:
`2021-11-17 16:09:12.259 ERROR [main][LocalizationImpl:889] java.lang.NullPointerException
java.lang.NullPointerException: null
	at com.ctc.wstx.sw.BaseStreamWriter.writeCharacters(BaseStreamWriter.java:498) ~[wstx.jar:3.2.8]
	at com.liferay.portal.util.LocalizationImpl.getXml(LocalizationImpl.java:876) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.util.LocalizationUtil.getXml(LocalizationUtil.java:322) ~[portal-kernel.jar:?]
	at com.liferay.commerce.inventory.internal.upgrade.v3_0_0.CommerceInventoryWarehouseUpgradeProcess.doUpgrade(CommerceInventoryWarehouseUpgradeProcess.java:61) ~[?:?]
	at com.liferay.portal.kernel.upgrade.UpgradeProcess.lambda$upgrade$0(UpgradeProcess.java:114) ~[portal-kernel.jar:?]
	at com.liferay.portal.db.partition.DBPartitionUtil.forEachCompanyId(DBPartitionUtil.java:113) [portal-impl.jar:?]
	at com.liferay.portal.dao.db.BaseDB.process(BaseDB.java:307) [portal-impl.jar:?]
	at com.liferay.portal.kernel.dao.db.BaseDBProcess.process(BaseDBProcess.java:241) [portal-kernel.jar:?]
	at com.liferay.portal.kernel.upgrade.UpgradeProcess.upgrade(UpgradeProcess.java:101) [portal-kernel.jar:?]
	at com.liferay.portal.kernel.upgrade.UpgradeProcess.upgrade(UpgradeProcess.java:138) [portal-kernel.jar:?]
	at com.liferay.portal.upgrade.internal.executor.UpgradeExecutor$UpgradeInfosRunnable.run(UpgradeExecutor.java:199) [bundleFile:?]
	at com.liferay.portal.output.stream.container.internal.OutputStreamContainerFactoryTrackerImpl.runWithSwappedLog(OutputStreamContainerFactoryTrackerImpl.java:134) [bundleFile:?]`

The issue is that the description is not present during the upgrade a NPE is thrown, so I'm just checking it and put an empty string a DB